### ISSUE TITLE
Formex + Money

### ISSUE THREAD

**Issue by coderberry** [jakub-zawislak/formex#13]
Hi. I love your library! It makes form building incredibly fun and easy.

One a project of mine, I'm using the [Money](https://hexdocs.pm/money/Money.html) library for some of my columns. For example:

```elixir
schema "campaigns" do
    field :name, :string
    field :bid_amount_cents, Money.Ecto.Type
  end
```

When I set up the type, it looks like this:

```elixir
defmodule  MyWeb.CampaignType do
  use Formex.Type

  def build_form(form) do
    form
    |> add(:name, :text_input, label: "Name", validation: [presence: true])
    |> add(:bid_amount_cents, :number_input, label: "Bid Amount", validation: [presence: true])
    |> add(:save, :submit, label: "Submit", phoenix_opts: [
      class: "btn-primary"
    ])
  end
end
```

My issue is that the form field that it renders does not use the amount value of the bid_amount_cents. Right now when it gets the value, it uses `@campaign.bid_amount_cents` when I need it to use `@campaign.bid_amount_cents.amount`. 

Is there a way to specify the value in the `CampaignType` above where I can derive the value as a function call? I assume this is possible with a custom field, but the docs are pretty shaky around custom fields and I can't find any good examples anywhere.

Thanks for your help!

**Comment by jakub-zawislak** [comment]
Please update to `0.6.3`. I have added a `custom_value` option:

```elixir
    |> add(:bid_amount_cents, :text_input, custom_value: fn struct ->
      # you may need this condition when using form collections
      if Map.get(struct, :bid_amount_cents) do
        Map.get(struct, :bid_amount_cents).amount
      else
        ""
      end
    end)
```

**Comment by jakub-zawislak** [comment]
Now I wonder if it was a good decision to pass a whole struct to this anonymous function. Maybe better would be to do so in that way:

```elixir
    |> add(:bid_amount_cents, :text_input, custom_value: fn value ->
      value && value.amount
    end)
```

or

```elixir
    |> add(:bid_amount_cents, :text_input, custom_value: fn value, _struct ->
      value && value.amount
    end)
```

**Comment by jakub-zawislak** [comment]
I chose the first option:
```elixir
    |> add(:bid_amount_cents, :text_input, custom_value: fn value ->
      value && value.amount
    end)
```
and published this in `0.6.4` version

**Comment by coderberry** [comment]
👏 Thank you!
